### ISSUE TITLE
ES5 Compatibility

### ISSUE THREAD

**Issue by TheBrokenRail** [svaarala/duktape#2053]
On the compatibility table, it is only 99% on ES5. This test is failing:

Enumerable properties can be shadowed by non-enumerables

**Comment by svaarala** [comment]
That's true, there are a few bugs/issues which affect some corner cases of compatibility.

But actually Duktape, and several other engines, are also no longer fully ES5.1 compatible because some core semantics follow later versions (ES2015+) which also redefine some ES5.1 behaviors. For example, Object built-in function argument coercion was changed after ES5.1, and Duktape tracks the later specifications rather than being strictly ES5.1. This is noted in https://duktape.org/guide.html#compatibility-e5.

What compatibility table are you referring to?

**Comment by TheBrokenRail** [comment]
http://kangax.github.io/compat-table/es5/

**Comment by svaarala** [comment]
You should file an issue there - basically they need a test which demonstrates the bug.
